pkg/cluster/generateconfig.go

```diff
@@ -28,143 +24,67 @@ import (
 	"github.com/openshift/installer/pkg/types/validation"
 	"golang.org/x/crypto/ssh"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/Azure/ARO-RP/pkg/api"
-	"github.com/Azure/ARO-RP/pkg/bootstraplogging"
-	"github.com/Azure/ARO-RP/pkg/cluster"
-	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
-	"github.com/Azure/ARO-RP/pkg/util/deployment"
+	"github.com/Azure/ARO-RP/pkg/api/validate"
 	"github.com/Azure/ARO-RP/pkg/util/pullsecret"
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
-func (m *manager) Create(ctx context.Context) error {
-	var err error
+func (m *manager) checkQuotas(ctx context.Context) error {
+	// we don't re-call Dynamic on subsequent entries here.  One reason is
+	// that we would re-check quota *after* we had deployed our VMs, and
+	// could fail with a false positive.
+	ocDynamicValidator := validate.NewOpenShiftClusterDynamicValidator(m.fpAuthorizer, m.log, m.env, m.doc.OpenShiftCluster, m.subscriptionDoc)
+	return ocDynamicValidator.Dynamic(ctx)
```

Comment:
hmm, I can see this has to be instantiated here, but it's ugly.  Should think about this as a follow-up.

Reply:
(don't attempt to solve this on this PR - too much change in one go)

Reply:
yeah :(